### PR TITLE
Remove outdated documentation things

### LINKED ISSUE #623: ledger-api docs link broken?
The Ledger API Introduction link looks broken here: https://github.com/digital-asset/daml/tree/master/ledger-api

### PR DESCRIPTION
**Pull request by bethaitman**
Fixes #623.

I've removed everything I know to be wrong from this README, which is the stuff about docs. But I'm pretty certain there's more wrong with it - for example, the stuff about releasing looks suspiciously out of date.

### CODE REVIEW

File: ledger-api/README.md [hunk 1]
```diff
@@ -38,27 +34,24 @@ sbt test
...
+2. Bump the version for the NPM distribution channel:
 
       (cd npm-package && npm version <major>.<minor>.<point>)
 
-4. Move the ## Unreleased section of [ledger-api/UNRELEASED.md](UNRELEASED.md) to the new release
+3. Move the ## Unreleased section of [ledger-api/UNRELEASED.md](UNRELEASED.md) to the new release
 version you have created in [ledger-api/CHANGELOG.md](CHANGELOG.md) and commit to that branch. The message of the commit is not important as it
 will be discarded;
-5. Run `sbt release`. This will ask for the release version and the next version, and will create
+4. Run `sbt release`. This will ask for the release version and the next version, and will create
```
Comment (gaboraranyossy-da) [inline ledger-api/README.md]: we don't use SBT anymore

File: ledger-api/README.md [hunk 2]
```diff
@@ -38,27 +34,24 @@ sbt test
...
 commit with the release version and next version, and also takes care about tagging;
-6. Push your **branch and tag**:
+5. Push your **branch and tag**:
 
 ```
 git push origin release/ledger-api/0.9.3        # check the tag that has been created, and push it
 git push -u origin ledger-api-release-0-9-3    # push your branch
 ```
-7. Go to the release [Jenkins job](http://ci.da-int.net/job/ledger-api/job/release/build?delay=0sec)
+6. Go to the release [Jenkins job](http://ci.da-int.net/job/ledger-api/job/release/build?delay=0sec)
```
Comment (gaboraranyossy-da) [inline ledger-api/README.md]: This is outdated too. We should probably link to our release procedure instead or just omit the release section in general.

Comment (bethaitman) [reply ledger-api/README.md]: I thought so. Where is the release procedure? If you give me a link I'll remove this section and sub it in instead.

Comment (gaboraranyossy-da) [reply ledger-api/README.md]: I am not sure where it is documented currently, but as we are doing head based development  the best would be I think is just removing the release section from here.

File: ledger-api/README.md (hunk 1 above)
Comment (bethaitman) [reply ledger-api/README.md]: Does this also apply to the `Build` section?

File: ledger-api/README.md (hunk 2 above)
Comment (bethaitman) [reply ledger-api/README.md]: Done!

File: ledger-api/README.md (hunk 1 above)
Comment (gaboraranyossy-da) [reply ledger-api/README.md]: yes

**Pull request merged**